From e036a5f01fbef907e00059aabcaf118e7e9630fb Mon Sep 17 00:00:00 2001
From: Paul Elliott <pelliott@blackpatchpanel.com>
Date: Wed, 11 Mar 2015 03:13:45 -0500
Subject: add manpages for simplescreenrecorder ssr-glinject

 E binary-without-manpage
 https://lintian.debian.org/tags/binary-without-manpage.html
 .
 Each binary in /usr/bin, /usr/sbin, /bin, /sbin or /usr/games should
 have a manual page
 .
 Note that though the man program has the capability to check for
 several program names in the NAMES section, each of these programs
 should have its own manual page (a symbolic link to the appropriate
 manual page is sufficient) because other manual page viewers such as
 xman or tkman don't support this.
 .
 If the name of the man page differs from the binary by case, man may be
 able to find it anyway; however, it is still best practice to make the
 case of the man page match the case of the binary.
 .
 If the man pages are provided by another package on which this package
 depends, lintian may not be able to determine that man pages are
 available. In this case, after confirming that all binaries do have man
 pages after this package and its dependencies are installed, please add
 a lintian override.
 .
 Refer to Debian Policy Manual section 12.1 (Manual pages) for details.
Forwarded: https://github.com/MaartenBaert/ssr/issues/318
---
 Makefile.am               |  4 ++++
 configure.ac              |  2 +-
 simplescreenrecorder.1    | 48 +++++++++++++++++++++++++++++++++++++++++++++++
 simplescreenrecorder.1.in | 48 +++++++++++++++++++++++++++++++++++++++++++++++
 ssr-glinject.1            | 44 +++++++++++++++++++++++++++++++++++++++++++
 ssr-glinject.1.in         | 44 +++++++++++++++++++++++++++++++++++++++++++
 6 files changed, 189 insertions(+), 1 deletion(-)
 create mode 100644 simplescreenrecorder.1
 create mode 100644 simplescreenrecorder.1.in
 create mode 100644 ssr-glinject.1
 create mode 100644 ssr-glinject.1.in

diff --git a/Makefile.am b/Makefile.am
index 4f7ae5a..b0c7ef6 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,5 +1,7 @@
 SUBDIRS = data glinject scripts src
 
+man1_MANS = simplescreenrecorder.1 ssr-glinject.1
+
 EXTRA_DIST = \
 	AUTHORS.md \
 	CHANGELOG.md \
@@ -9,6 +11,8 @@ EXTRA_DIST = \
 	README.md \
 	simple-build-and-install \
 	simple-uninstall \
+	simplescreenrecorder.1.in simplescreenrecorder.1 \
+	ssr-glinject.1.in ssr-glinject.1 \
 	todo.txt
 
 ACLOCAL_AMFLAGS = -I m4
diff --git a/configure.ac b/configure.ac
index 50426fe..f817f05 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3,7 +3,7 @@ AC_CONFIG_SRCDIR([src/Main.cpp])
 AC_CONFIG_AUX_DIR([build-aux])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_HEADERS([config.h])
-AC_CONFIG_FILES([Makefile data/Makefile glinject/Makefile scripts/Makefile src/Makefile])
+AC_CONFIG_FILES([Makefile data/Makefile glinject/Makefile scripts/Makefile src/Makefile simplescreenrecorder.1 ssr-glinject.1])
 
 AM_INIT_AUTOMAKE([foreign subdir-objects -Wall -Werror])
 AM_MAINTAINER_MODE
diff --git a/simplescreenrecorder.1 b/simplescreenrecorder.1
new file mode 100644
index 0000000..31de2f1
--- /dev/null
+++ b/simplescreenrecorder.1
@@ -0,0 +1,48 @@
+.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.46.4.
+.TH SIMPLESCREENRECORDER "1" "March 2015" "simplescreenrecorder 0.3.3" "User Commands"
+.SH NAME
+simplescreenrecorder \- X11 screen recorder. version 0.3.3
+.SH SYNOPSIS
+.B simplescreenrecorder
+[\fI\,OPTIONS\/\fR]
+.SH DESCRIPTION
+.B simplescreenrecorder
+is a feature-rich screen recorder that supports X11 and OpenGL. It has
+a Qt\-based graphical user interface. It can record the entire screen
+or part of it, or record OpenGL applications directly. The recording
+can be paused and resumed at any time. Many different file formats and
+codecs are supported.
+.SH OPTIONS
+.TP
+\fB\-\-help\fR
+Show this help message.
+.TP
+\fB\-\-logfile\fR
+Write log to ~/.ssr/log\-DATE_TIME.txt instead of stdout.
+.TP
+\fB\-\-statsfile\fR[=\fI\,FILE\/\fR]
+Write recording statistics to FILE. If FILE is omitted,
+\fI\,/dev/shm/simplescreenrecorder\-stats\-PID\/\fP is used. It will
+be updated continuously and deleted when the recording
+page is closed.
+.TP
+\fB\-\-syncdiagram\fR
+Show synchronization diagram (for debugging).
+.TP
+\fB\-\-no\-systray\fR
+Don't show the system tray icon.
+.TP
+\fB\-\-start\-hidden\fR
+Start the application in hidden form.
+.SH "SEE ALSO"
+The full documentation for
+.B simplescreenrecorder
+is maintained as a Texinfo manual.  If the
+.B info
+and
+.B simplescreenrecorder
+programs are properly installed at your site, the command
+.IP
+.B info simplescreenrecorder
+.PP
+should give you access to the complete manual.
diff --git a/simplescreenrecorder.1.in b/simplescreenrecorder.1.in
new file mode 100644
index 0000000..eceb1b2
--- /dev/null
+++ b/simplescreenrecorder.1.in
@@ -0,0 +1,48 @@
+.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.46.4.
+.TH SIMPLESCREENRECORDER "1" "March 2015" "simplescreenrecorder @PACKAGE_VERSION@" "User Commands"
+.SH NAME
+simplescreenrecorder \- X11 screen recorder. version @PACKAGE_VERSION@
+.SH SYNOPSIS
+.B simplescreenrecorder
+[\fI\,OPTIONS\/\fR]
+.SH DESCRIPTION
+.B simplescreenrecorder
+is a feature-rich screen recorder that supports X11 and OpenGL. It has
+a Qt\-based graphical user interface. It can record the entire screen
+or part of it, or record OpenGL applications directly. The recording
+can be paused and resumed at any time. Many different file formats and
+codecs are supported.
+.SH OPTIONS
+.TP
+\fB\-\-help\fR
+Show this help message.
+.TP
+\fB\-\-logfile\fR
+Write log to ~/.ssr/log\-DATE_TIME.txt instead of stdout.
+.TP
+\fB\-\-statsfile\fR[=\fI\,FILE\/\fR]
+Write recording statistics to FILE. If FILE is omitted,
+\fI\,/dev/shm/simplescreenrecorder\-stats\-PID\/\fP is used. It will
+be updated continuously and deleted when the recording
+page is closed.
+.TP
+\fB\-\-syncdiagram\fR
+Show synchronization diagram (for debugging).
+.TP
+\fB\-\-no\-systray\fR
+Don't show the system tray icon.
+.TP
+\fB\-\-start\-hidden\fR
+Start the application in hidden form.
+.SH "SEE ALSO"
+The full documentation for
+.B simplescreenrecorder
+is maintained as a Texinfo manual.  If the
+.B info
+and
+.B simplescreenrecorder
+programs are properly installed at your site, the command
+.IP
+.B info simplescreenrecorder
+.PP
+should give you access to the complete manual.
diff --git a/ssr-glinject.1 b/ssr-glinject.1
new file mode 100644
index 0000000..81b5d23
--- /dev/null
+++ b/ssr-glinject.1
@@ -0,0 +1,44 @@
+.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.46.4.
+.TH SSR-GLINJECT "1" "March 2015" "ssr-glinject 0.3.3" "User Commands"
+.SH NAME
+ssr-glinject \- GLX injection tool. version 0.3.3
+.SH SYNOPSIS
+.B ssr-glinject
+[\fI\,OPTIONS\/\fR] [\fI\,--\/\fR] \fI\,COMMAND\/\fR
+.SH OPTIONS
+.TP
+\fB\-\-help\fR
+Show this help message.
+.TP
+\fB\-\-glx\-debug\fR
+Enables GLX debugging. This may reduce the performance
+and print lots of error messages, but it is useful to
+track down bugs.
+.TP
+\fB\-\-relax\-permissions\fR
+Uses mode 666 instead of 600 for shared memory, so that
+other users can record the stream. This is insecure and
+should not be used on a computer that can be accessed by
+other users that you don't trust.
+.TP
+\fB\-\-channel\fR=\fI\,CHANNEL\/\fR
+Channel name to use. The default is 'channel\-USERNAME'.
+.PP
+This script uses LD_PRELOAD to inject the GLInject library into the given
+command, so that SimpleScreenRecorder can record it. It should be safe to use
+this on all applications (including command\-line programs and shell scripts).
+If the program doesn't use OpenGL, it should have no effect. If you find a
+program that crashes or behaves incorrectly when GLInject is used, please submit
+a bug report.
+.SH "SEE ALSO"
+The full documentation for
+.B ssr-glinject
+is maintained as a Texinfo manual.  If the
+.B info
+and
+.B ssr-glinject
+programs are properly installed at your site, the command
+.IP
+.B info ssr-glinject
+.PP
+should give you access to the complete manual.
diff --git a/ssr-glinject.1.in b/ssr-glinject.1.in
new file mode 100644
index 0000000..87f361a
--- /dev/null
+++ b/ssr-glinject.1.in
@@ -0,0 +1,44 @@
+.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.46.4.
+.TH SSR-GLINJECT "1" "March 2015" "ssr-glinject @PACKAGE_VERSION@" "User Commands"
+.SH NAME
+ssr-glinject \- GLX injection tool. version @PACKAGE_VERSION@
+.SH SYNOPSIS
+.B ssr-glinject
+[\fI\,OPTIONS\/\fR] [\fI\,--\/\fR] \fI\,COMMAND\/\fR
+.SH OPTIONS
+.TP
+\fB\-\-help\fR
+Show this help message.
+.TP
+\fB\-\-glx\-debug\fR
+Enables GLX debugging. This may reduce the performance
+and print lots of error messages, but it is useful to
+track down bugs.
+.TP
+\fB\-\-relax\-permissions\fR
+Uses mode 666 instead of 600 for shared memory, so that
+other users can record the stream. This is insecure and
+should not be used on a computer that can be accessed by
+other users that you don't trust.
+.TP
+\fB\-\-channel\fR=\fI\,CHANNEL\/\fR
+Channel name to use. The default is 'channel\-USERNAME'.
+.PP
+This script uses LD_PRELOAD to inject the GLInject library into the given
+command, so that SimpleScreenRecorder can record it. It should be safe to use
+this on all applications (including command\-line programs and shell scripts).
+If the program doesn't use OpenGL, it should have no effect. If you find a
+program that crashes or behaves incorrectly when GLInject is used, please submit
+a bug report.
+.SH "SEE ALSO"
+The full documentation for
+.B ssr-glinject
+is maintained as a Texinfo manual.  If the
+.B info
+and
+.B ssr-glinject
+programs are properly installed at your site, the command
+.IP
+.B info ssr-glinject
+.PP
+should give you access to the complete manual.
